addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/gui/PopupMenuParamMiner.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer.gui;
+
+import org.parosproxy.paros.model.SiteNode;
+import org.zaproxy.addon.paramminer.ExtensionParamMiner;
+import org.zaproxy.zap.view.popup.PopupMenuItemSiteNodeContainer;
+
+public class PopupMenuParamMiner extends PopupMenuItemSiteNodeContainer {
+
+    private static final long serialVersionUID = 1L;
+    private ExtensionParamMiner extension;
+
+    public PopupMenuParamMiner(ExtensionParamMiner ext, String label) {
+        super(label);
+        this.extension = ext;
+    }
+
+    @Override
+    protected void performAction(SiteNode node) {
+        if (node != null) {
```

Comment:
out of interest, when can this be null?

Reply:
when I'm starting param miner from scratch without anything other than URL?

Reply:
Wont you still want to display the dialog then?

Reply:
It is being displayed :)

Reply:
Wait ... Then the same applies for AjaxSpider :)

Reply:
The `node` is never `null` here. (If you check the JavaDoc of `performAction` it states that.)

Reply:
Okay so I can safely remove the null check then